### DISCUSSION TITLE
Admin Page not usable after Update

### DISCUSSION THREAD

**Discussion by SchnitzelDude1337** [dani-garcia/vaultwarden#4638] [Q&A] [answered]
Hey,

yesterday i updated a Vaultwarden 1.23.1 to a 1.30.5. I know, big version jump, but i often read that a bigger version jump would be no problem for Vaultwarden and so far everything seems to work fine, especially after i set the correct variables. However, later i discovered a problem: i can't really use my Admin Page So, the "ADMIN_TOKEN" variable is set and i can reach it, but it looks like this:

![VaultwardenAdminProblem](https://github.com/dani-garcia/vaultwarden/assets/172364770/6b3291d1-82c1-47eb-ab15-0204fc764f2c)

and i also get the following error in my log files:

`[2024-06-11 06:40:40.975][error][ERROR] Handlebars.
[CAUSE] RenderError {
    template_name: Some(
        "admin/settings",
    ),
    line_no: Some(
        12,
    ),
    column_no: Some(
        17,
    ),
    reason: MissingVariable(
        Some(
            "config",
        ),
    ),
    unimplemented: false,
}
`

I definitely did not touch the templates and pretty much did a standard upgrade of Vaultwarden. I also have a backup of my "data" Folder before the upgrade, if i need anything from these files and did a quick comparison between the two "admin/settings" files, and nothing out of the ordinary.

Any advice on what could be problem?

**Comment by stefan0xC** [comment]
The handlebar template have changed. Cf. https://github.com/dani-garcia/vaultwarden/issues/4610#issuecomment-2147023971 (and given the error message probably even more). So your custom template files have to be updated (with [every relevant change](https://github.com/dani-garcia/vaultwarden/commits/1.30.5/src/static/templates/admin) between 1.23.1 and 1.30.5) accordingly.

**Reply by SchnitzelDude1337** [reply]
Ok alright, much appreciated and thanks already!

Since i am not really well versed in Handlebar or anything regarding these templates: how exactly do i update these files "accordingly"?

**Reply by stefan0xC** [reply, marked as answer]
I don't know which files you have adapted but I recommend that you check with `git diff 1.23.1...1.30.5 -- src/static/templates/admin/` what has changed and apply the differences. Or if it's easier, you might do it the other way, copy the [templates from `1.30.5`](https://github.com/dani-garcia/vaultwarden/tree/1.30.5/src/static/templates/admin) and reapply your changes.

**Reply by SchnitzelDude1337** [reply]
Thanks! That was it!

Have a great day :)
